core/trino-main/src/main/java/io/trino/operator/output/PositionsAppenderPageBuilder.java

```diff
@@ -99,7 +99,9 @@ public boolean isEmpty()
     public Page build()
     {
         if (channelAppenders.length == 0) {
```

Comment:
It would be good to have a test for this case (I can add it in a separate PR). Do you know what kind of query has zero-column partitioning?

Reply:
> It would be good to have a test for this case (I can add it in a separate PR). Do you know what kind of query has zero-column partitioning?

I observed that with queries like this one: 
```
SELECT o1.orderkey FROM orders o1 RIGHT JOIN (SELECT * FROM orders LIMIT 10) o2 ON true ORDER BY o1.orderkey LIMIT 20
```

where you are joining two tables, where neither join condition, nor output depend on any column from one of the tables.

As for the testing, it is problematic as I only observed problem when `POO.finish()` was called twice. And this does not happen often.